hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaCache.java

```diff
@@ -74,12 +83,15 @@ public class QuotaCache implements Stoppable {
   private final ConcurrentHashMap<TableName, Double> tableMachineQuotaFactors =
     new ConcurrentHashMap<>();
   private final RegionServerServices rsServices;
+  private final String userOverrideRequestAttributeKey;
 
   private QuotaRefresherChore refreshChore;
   private boolean stopped = true;
 
   public QuotaCache(final RegionServerServices rsServices) {
     this.rsServices = rsServices;
+    this.userOverrideRequestAttributeKey = rsServices.getConfiguration()
+      .get(QUOTA_USER_REQUEST_ATTRIBUTE_OVERRIDE_KEY, QUOTA_USER_REQUEST_ATTRIBUTE_OVERRIDE_KEY);
```

Comment:
By default you can just add the request attribute `hbase.quota.user.override.key` and use this feature, so you aren't _required_ to update the configuration

Reply:
Oh, I see that you do in fact use this to look up what the configuration key should be. In that case, I think the default key _value_ should be `hbase.quota.user.override`, without the `.key` suffix, so that it aligns with our usual configuration naming scheme.

Reply:
I worry a bit that this could mislead people into thinking that the value of `hbase.quota.user.override` should be their override, rather than the key to be provided as a request attribute. Whereas I think `hbase.quota.user.override.key` is more clearly not that

Reply:
Hmm yeah okay. Maybe I'm also suffering from a parsing error from "user.override.key" vs. "user_override_key" -- we're a bit lazy with how we build configuration keys because there's no explicit hierarchy tied to the `.` separator.

Reply:
To me it makes sense for `hbase.quota.user.override.key -> hbase.quota.user.override`.  So the user would pass an attribute `hbase.quota.user.override -> foo`

But how about we just default to empty and skip the debate? Personally I think this is a very long attribute name.

Reply:
Sounds good to me, I'll rip out the default. Makes the testing simpler too